file/builder.go

```diff
@@ -63,6 +65,10 @@ func (b *stateBuilder) build() (*utils.KongRawState, *utils.KonnectRawState, err
 	}
 	b.defaulter = defaulter
 
+	if utils.Kong300Version.LTE(b.kongVersion) {
+		b.checkRoutePaths = true
+	}
+
```

Comment:
why not move this function into builder::ingestRoute()? This would remove the condition check here and remove the `checkRoutePaths` variable from the stateBuilder.

Reply:
The reason for moving it here was to avoid doing version comparison for all the routes being ingested (addressing [this comment](https://github.com/Kong/deck/pull/753#discussion_r971336120))